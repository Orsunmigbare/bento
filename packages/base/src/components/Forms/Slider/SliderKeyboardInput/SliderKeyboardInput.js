import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { linear } from "easing-utils"
import { inverseLerp, clamp } from "@staccx/math"
import withTheme from "../../../../utils/withTheme"
import Input from "../../Input/Input"
import Slider from "../Slider/Slider"

const SliderWrapper = withTheme(
  styled.div`
    ${props => props.themeStyle(props)};
  `,
  "SliderKeyboardInput.SliderWrapper"
)

const HiddenLabel = withTheme(
  styled.label`
    ${props => props.themeStyle(props)};
  `,
  "SliderKeyboardInput.HiddenLabel"
)

const SliderInput = withTheme(
  styled(Input)`
    ${props => props.themeStyle(props)};
  `,
  "SliderKeyboardInput.SliderInput"
)

class SliderKeyboardInput extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isAnimatingIn: false,
      currentValue: 0,
      percentage: 0
    }
    this.handleChange = this.handleChange.bind(this)
    this.countUp = this.countUp.bind(this)
    this.stopCount = this.stopCount.bind(this)
    this.updateState = this.updateState.bind(this)
    this.tick = 0
  }

  componentDidMount() {
    if (this.props.easingFunction) {
      this.startCount()
    }
  }

  componentWillUnmount() {
    if (this.props.easingFunction) {
      this.stopCount()
    }
  }

  startCount() {
    if (!this._frameId) {
      this.setState(
        {
          isAnimatingIn: true
        },
        () => {
          this._frameId = window.requestAnimationFrame(this.countUp)
        }
      )
    }
  }

  countUp() {
    // perform loop work here
    if (this.tick < this.props.animationTicks && this.props.easingFunction) {
      const t = this.props.easingFunction(this.tick / this.props.animationTicks)
      const newValue = this.props.value * t

      this.updateState(newValue).then(() => {
        this._frameId = window.requestAnimationFrame(this.countUp)
      })
    } else {
      this.updateState(this.props.value, {
        isAnimatingIn: false
      }).then(this.stopCount)
    }
    this.tick++
  }

  stopCount() {
    window.cancelAnimationFrame(this._frameId)
  }

  updateState(value, options = {}) {
    return new Promise(resolve => {
      this.setState(
        {
          currentValue: clamp(this.props.min, this.props.max, Math.ceil(value)),
          percentage: inverseLerp(this.props.min, this.props.max, value) * 100,
          ...options
        },
        resolve
      )
    })
  }

  handleChange(event) {
    const value = event.target.value
    this.updateState(value).then(() => {
      if (this.props.onChange) {
        this.props.onChange(this.state.currentValue)
      }
    })
  }

  render() {
    const {
      label,
      name,
      min,
      max,
      step,
      mask,
      className,
      themeVariant
    } = this.props
    const { currentValue, percentage } = this.state
    return (
      <SliderWrapper className={className} themeVariant={themeVariant}>
        <HiddenLabel htmlFor={name} themeVariant={themeVariant}>
          {label}
        </HiddenLabel>
        <SliderInput
          type="text"
          name={`${name}-keyboard`}
          id={`${name}-keyboard`}
          value={currentValue}
          onChange={this.handleChange}
          min={min}
          max={max}
          mask={mask || null}
          disabled={this.state.isAnimatingIn}
          themeVariant={themeVariant}
        />
        <Slider
          percentage={percentage}
          value={currentValue}
          name={name}
          id={name}
          step={step}
          min={min}
          max={max}
          onChange={this.handleChange}
        />
      </SliderWrapper>
    )
  }
}

SliderKeyboardInput.defaultProps = {
  value: 0,
  min: 0,
  max: 100,
  step: 1,
  animationTicks: 50,
  easingFunction: linear,
  onChange: null,
  mask: null,
  className: ""
}

SliderKeyboardInput.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.number,
  min: PropTypes.number,
  max: PropTypes.number,
  onChange: PropTypes.func,
  step: PropTypes.number,
  animationTicks: PropTypes.number,
  easingFunction: PropTypes.func,
  mask: PropTypes.array,
  className: PropTypes.string
}

export default withTheme(SliderKeyboardInput)
