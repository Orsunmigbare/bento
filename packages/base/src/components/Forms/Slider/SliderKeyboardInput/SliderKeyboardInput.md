```jsx
<div>
  <SliderKeyboardInput
    name="theSlider"
    label="Accessibility label"
    onChange={() => console.log("Slide with input")}
  />
</div>
```
