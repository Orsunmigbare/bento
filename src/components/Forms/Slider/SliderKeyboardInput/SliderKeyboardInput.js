import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"
import { linear } from "easing-utils"
import { inverseLerp, clamp } from "../../../../math"
import Label from "../../Label/Label"
import Input from "../../Input/Input"
import Slider from "../Slider/Slider"
import {
  applyVariants,
  targetSize,
  fontFamily,
  font,
  color
} from "../../../../theming"
import themeProps from "./SliderKeyboardInput.themeProps"

const removeSpaces = value => value.replace(/\s/g, "")

class SliderKeyboardInput extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isAnimatingIn: false,
      currentValue: 0,
      percentage: 0
    }
    this.handleBlur = this.handleBlur.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.countUp = this.countUp.bind(this)
    this.stopCount = this.stopCount.bind(this)
    this.updateState = this.updateState.bind(this)
    this.tick = 0
  }

  componentDidMount() {
    if (this.props.easingFunction) {
      this.startCount()
    }
  }

  componentWillUnmount() {
    if (this.props.easingFunction) {
      this.stopCount()
    }
  }

  startCount() {
    if (!this._frameId) {
      this.setState(
        {
          isAnimatingIn: true
        },
        () => {
          this._frameId = window.requestAnimationFrame(this.countUp)
        }
      )
    }
  }

  countUp() {
    // perform loop work here
    if (this.tick < this.props.animationTicks && this.props.easingFunction) {
      const t = this.props.easingFunction(this.tick / this.props.animationTicks)
      const newValue = this.props.value * t

      this.updateState(newValue).then(() => {
        this._frameId = window.requestAnimationFrame(this.countUp)
      })
    } else {
      this.updateState(this.props.value, {
        isAnimatingIn: false
      }).then(this.stopCount)
    }
    this.tick++
  }

  stopCount() {
    window.cancelAnimationFrame(this._frameId)
  }

  updateState(value, options = {}) {
    return new Promise(resolve => {
      this.setState(
        {
          currentValue: Math.floor(value),
          percentage:
            inverseLerp(
              this.props.min,
              this.props.max,
              clamp(this.props.min, this.props.max, Math.ceil(value))
            ) * 100,
          ...options
        },
        resolve
      )
    })
  }

  handleChange(event) {
    const value = event.target.value
    this.updateState(parseInt(removeSpaces(value), 10)).then(() => {
      if (this.props.onChange) {
        this.props.onChange(this.state.currentValue)
      }
    })
  }

  handleBlur(event) {
    const value = event.target.value
    const valueInt = parseInt(removeSpaces(value), 10)
    const parsed = clamp(this.props.min, this.props.max, valueInt)
    this.updateState(parsed).then(() => {
      if (this.props.onBlur) {
        this.props.onBlur(this.state.currentValue)
      }
    })
  }

  render() {
    const {
      label,
      name,
      min,
      max,
      step,
      mask,
      className,
      ignoreBase,
      variant,
      showLabel
    } = this.props
    const { currentValue, percentage } = this.state
    return (
      <SliderWrapper
        className={className}
        variant={variant}
        ignoreBase={ignoreBase}
      >
        <Label htmlFor={name} variant={variant}>
          {label}
        </Label>
        <SliderInput
          type="text"
          name={`${name}-keyboard`}
          id={`${name}-keyboard`}
          value={currentValue}
          onChange={this.handleChange}
          onBlur={this.handleBlur}
          label={showLabel ? label : null}
          min={min}
          max={max}
          mask={mask || null}
          disabled={this.state.isAnimatingIn}
          variant={variant}
        />
        <Slider
          percentage={percentage}
          value={currentValue}
          name={name}
          id={name}
          step={step}
          min={min}
          max={max}
          onChange={this.handleChange}
          variant={variant}
        />
      </SliderWrapper>
    )
  }
}

const SliderWrapper = styled.div`
  font-family: ${fontFamily.body};
  font-size: ${font.input};
  ${applyVariants(SliderKeyboardInput.themeProps.wrapper)};
`

const SliderInput = styled(Input)`
  background-color: transparent;
  border: 0;
  min-height: ${targetSize.normal};
  max-width: 100%;
  -moz-appearance: textfield;
  outline: none;

  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  &:hover,
  &:active,
  &:focus,
  &:-moz-ui-invalid,
  &:invalid {
    outline: none;
    box-shadow: none;
  }

  &:disabled {
    color: ${color.text};
  }
  ${applyVariants(SliderKeyboardInput.themeProps.input)};
`

SliderKeyboardInput.defaultProps = {
  animationTicks: 50,
  className: "",
  easingFunction: linear,
  mask: null,
  max: 100,
  min: 0,
  onChange: null,
  step: 1,
  value: 0,
  showLabel: false
}

SliderKeyboardInput.propTypes = {
  animationTicks: PropTypes.number,
  className: PropTypes.string,
  easingFunction: PropTypes.func,
  ignoreBase: PropTypes.bool,
  label: PropTypes.string.isRequired,
  mask: PropTypes.func,
  max: PropTypes.number,
  min: PropTypes.number,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  step: PropTypes.number,
  variant: PropTypes.string,
  value: PropTypes.number,
  showLabel: PropTypes.bool
}
SliderKeyboardInput.themeProps = themeProps

export default SliderKeyboardInput
