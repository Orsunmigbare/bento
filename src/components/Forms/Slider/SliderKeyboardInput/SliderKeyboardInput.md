```jsx
<div>
  <SliderKeyboardInput
    name="theSlider"
    label="Accessibility label"
    onChange={() => null}
  />
</div>
```

```jsx
<div>
  <SliderKeyboardInput
    name="theSlider"
    label="Accessibility label"
    onChange={() => null}
    min={0}
    max={100000}
  />
</div>
```


