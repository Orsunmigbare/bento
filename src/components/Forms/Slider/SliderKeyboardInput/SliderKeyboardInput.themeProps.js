import { themePropTypes } from "../../../../theming"

const themeProps = {
  wrapper: {
    name: "SLIDER_KEYBOARD_WRAPPER",
    description: "Wrapper style",
    type: themePropTypes.style
  },
  input: {
    name: "SLIDER_KEYBOARD_INPUT",
    description: "Wrapper style",
    type: themePropTypes.style
  }
}

export default themeProps
