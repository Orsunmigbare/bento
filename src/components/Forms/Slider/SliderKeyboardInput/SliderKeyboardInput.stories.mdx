import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import SliderKeyboardInput from "./SliderKeyboardInput"

<Meta title="Components|Forms/SliderKeyboardInput" component={SliderKeyboardInput}/>


# Slider


```javascript
import {SliderKeyboardInput} from "@staccx/bento"
```

## Example usage
<Preview>
  <Story name="Example">
    <SliderKeyboardInput label="Slider with keyboard input" min={100} max={500} step={50} defaultValue={250} onChange={(value) => console.log("The value is ",value)}  />
  </Story>
</Preview>

## Props for Slider
<Props of={SliderKeyboardInput} />
