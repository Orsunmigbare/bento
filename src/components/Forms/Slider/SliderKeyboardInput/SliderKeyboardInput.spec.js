import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import SliderKeyboardInput from "./SliderKeyboardInput"

describe("SliderKeyboardInput", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Name", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput name="slider" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("showLabel", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput showLabel />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("value", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput value={100} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("step", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput step={1} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("onBlur", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput onBlur={() => console.log("test")} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("onChange", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput onChange={() => console.log("test")} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("min", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput min={0} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("max", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput max={1000} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("label", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput label="Label" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("all", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <SliderKeyboardInput
              showLabel
              step={1}
              value={100}
              name="slider"
              label="Label"
              onBlur={() => console.log("test")}
              onChange={e => console.log(e.target.value)}
              max={1000}
              min={0}
            />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
