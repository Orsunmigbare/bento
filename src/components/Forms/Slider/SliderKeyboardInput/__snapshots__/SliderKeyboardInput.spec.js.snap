// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderKeyboardInput Snapshots Name 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="slider-keyboard"
      max={100}
      min={0}
      name="slider-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
      value={NaN}
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="sc-AykKK fQxNoe sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO hqHBeH"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": NaN,
              "value": NaN,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots No props 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-AykKK glxnfR sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO jzbrdi"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots all 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    <label
      className="sc-AykKC bonmue sc-AykKH hciGUB"
      htmlFor="slider-keyboard"
    >
      Label
      
    </label>
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="slider-keyboard"
      max={1000}
      min={0}
      name="slider-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={1000}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-AykKK glxnfR sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO jzbrdi"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots label 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    <label
      className="sc-AykKC bonmue sc-AykKH hciGUB"
      htmlFor="undefined-keyboard"
    >
      Label
      
    </label>
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
      value={NaN}
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="sc-AykKK fQxNoe sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO hqHBeH"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": NaN,
              "value": NaN,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots max 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={1000}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={1000}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-AykKK glxnfR sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO jzbrdi"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots min 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
      value={NaN}
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="sc-AykKK fQxNoe sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO hqHBeH"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": NaN,
              "value": NaN,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots onBlur 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
      value={NaN}
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="sc-AykKK fQxNoe sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO hqHBeH"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": NaN,
              "value": NaN,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots onChange 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-AykKK glxnfR sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO jzbrdi"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots showLabel 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-AykKK glxnfR sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO jzbrdi"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots step 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="sc-AykKK glxnfR sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO jzbrdi"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": 0,
              "value": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SliderKeyboardInput Snapshots value 1`] = `
<div
  className="sc-fzXfLU dsFUbL"
>
  <div
    className="sc-AykKI iDYrmt sc-fzXfLV hicBTo"
  >
    
    
    <input
      autoFocus={null}
      className="sc-AykKJ gjiytr"
      disabled={false}
      hidden={false}
      id="undefined-keyboard"
      max={100}
      min={0}
      name="undefined-keyboard"
      onBlur={null}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={null}
      placeholder={null}
      style={null}
      type="text"
      value={NaN}
    />
  </div>
  <div
    className="sc-fzXfLP TLbY"
  >
    <div
      className="sc-fzXfLQ kuAMaB"
      style={Object {}}
    >
      <div
        className="sc-fzXfLR gMdHUN"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      />
      <div
        className="slider-handles"
      >
        <button
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="sc-AykKK fQxNoe sc-fzXfLS djxtxC"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
          type="button"
        />
      </div>
      <div
        className="sc-fzXfLT gJDTEr"
      >
        <div
          className="sc-fzXfLO hqHBeH"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          target={
            Object {
              "id": "$$-0",
              "percent": NaN,
              "value": NaN,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
